src/cfml/system/endpoints/S3.cfc

```diff
@@ -0,0 +1,193 @@
+
+    public string function resolvePackage(required string package, boolean verbose=false) {
+        var bucket = package.listFirst('/');
+        var objectKey = package.listRest('/');
+        var awsSettings = resolveAwsSettings();
+        var presignedPath = generatePresignedPath(bucket, objectKey, awsSettings);
+        return httpsEndpoint.resolvePackage(presignedPath, verbose);
+    }
+
+    public function getDefaultName(required string package) {
+        return httpsEndpoint.getDefaultName(package);
+    }
+
+    public function getUpdate(required string package, required string version, boolean verbose=false) {
+        return httpsEndpoint.getUpdate(argumentCollection = arguments);
+    }
+
+    private function generatePresignedPath(bucket, objectKey, awsSettings) {
+        var isoTime = iso8601();
+        var bucketRegion = resolveBucketRegion(bucket, awsSettings.defaultRegion);
+        var host = 's3.#bucketRegion#.amazonaws.com';
+        var path = encodeUrl('/#bucket#/#objectKey#', false);
+
+        // query string
+        var qs = {
+            'X-Amz-Algorithm': 'AWS4-HMAC-SHA256',
+            'X-Amz-Credential': awsSettings.awsKey & '/' & isoTime.left( 8 ) & '/' & bucketRegion & '/s3/aws4_request',
+            'X-Amz-Date': isoTime,
+            'X-Amz-Expires': 300,
+            'X-Amz-SignedHeaders': 'host'
+        };
+        if (awsSettings.sessionToken.len()) {
+            qs['X-Amz-Security-Token'] = awsSettings.sessionToken;
+        }
+        qs = qs.keyArray()
+            .sort('text')
+            .reduce((r, key) => r.listAppend('#key#=#encodeUrl(qs[key])#', '&'), '');
+
+        // canonical request
+        var canonicalRequest = [
+            'GET',
+            path,
+            qs,
+            'host:#host#',
+            '',
+            'host',
+            'UNSIGNED-PAYLOAD'
+        ].toList(chr(10));
+
+        // string to sign
+        var stringtoSign = [
+            'AWS4-HMAC-SHA256',
+            isoTime,
+            isoTime.left( 8 ) & '/' & bucketRegion & '/s3/aws4_request',
+            hash(canonicalRequest, 'SHA-256').lcase()
+        ].toList(chr(10));
+
+        // signature
+        var signingKey = binaryDecode(hmac(isoTime.left(8), 'AWS4' & awsSettings.awsSecretKey, 'hmacSHA256', 'utf-8'), 'hex');
+        signingKey = binaryDecode(hmac(bucketRegion, signingKey, 'hmacSHA256', 'utf-8'), 'hex');
+        signingKey = binaryDecode(hmac('s3', signingKey, 'hmacSHA256', 'utf-8'), 'hex');
+        signingKey = binaryDecode(hmac('aws4_request', signingKey, 'hmacSHA256', 'utf-8'), 'hex');
+        var signature = hmac(stringToSign, signingKey, 'hmacSHA256', 'utf-8').lcase();
+
+        qs &= '&X-Amz-Signature=' & signature;
+        return '//' & host & path & '?' & qs;
+    }
+
+    private function resolveBucketRegion(bucket, defaultRegion) {
+        var req = '';
+        cfhttp(url='https://s3.#defaultRegion#.amazonaws.com', method='HEAD', result='req', redirect=false) {
```

Comment:
CFHTTP calls should probably use any proxy settings configured in CommandBox.  See here:
https://github.com/Ortus-Solutions/commandbox/blob/development/src/cfml/system/util/ForgeBox.cfc#L485-L508